java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/ShareFileController.java

```diff
@@ -60,31 +65,40 @@ public ResponseEntity<String> generateLink(@RequestParam String fileName,
 		if (userFile == null) {
 			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
 		}
-		String code = shareGpxService.generateSharedCode(userFile, type, groupActions);
+		String code = shareFileService.generateSharedCode(userFile, type, groupActions);
+		if (code == null) {
+			return ResponseEntity.badRequest().body("Error generating link");
+		}
+		String name = shareFileService.getNamePartForCode(fileName);
+		String url = "/share/get?code=" + code + "-" + name;
```

Comment:
Seems a little awkward. API's parameter code is not exactly code but share the same name. Better to map /share/get/code/filename, where /filename could be optional. IMHO.